nameserver/mocks_test.go

```diff
@@ -75,6 +75,10 @@ func (mz *mockedZoneWithRecords) AddRecord(ident string, name string, ip net.IP)
 	notImplWarn()
 	return nil
 }
+func (mz *mockedZoneWithRecords) DeleteRecords(ident *string, ip *net.IP, name *string) int {
```

Comment:
There is something I don't like: `DeleteRecords` takes **pointers** to strings/IPs, while `DeleteRecord` does not. I understand it is for passing `nil` as a _wildcard argument_, but maybe we could use the empty values for that (eg, `ip=net.IP{}`)...